ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/NoUnusedImportsRule.kt

```diff
@@ -112,5 +126,43 @@ class NoUnusedImportsRule : Rule("no-unused-imports") {
         }
     }
 
+    // Builds a list of imports. Takes care of having aliases in case it is assigned to imports
+    private fun buildNonRedundantImports(text: String) {
+        val element = text.split("import").last().trim()
+        if (element.contains(" as ")) {
```

Comment:
there is an ElementType for the `as` keyword: `~.c.i.p.impl.source.tree.LeafPsiElement (AS_KEYWORD) "as"`

Is it possible to check that instead? Seems slightly more robust than text matching.

Reply:
That would help. But in this function, the import is sent as a String. The presence of " as " keyword comes in the subsequent node visits.